source/AI.cpp

```diff
@@ -324,6 +324,16 @@ void AI::IssueShipTarget(const PlayerInfo &player, const shared_ptr<Ship> &targe
 
 
 
+void AI::IssueAsteroidTarget(const PlayerInfo &player, const shared_ptr<Minable> &targetAsteroid)
+{
+	Orders newOrders;
+	newOrders.type = Orders::MINING;
+	newOrders.targetAsteroid = targetAsteroid;
+	IssueOrders(player, newOrders, "focusing fire on " + targetAsteroid->Name() + " asteroid.");
```

Comment:
A minable is not required to be an "asteroid;" that just happens to be true for the base game.

We should probably add support for `Body::noun` and let the various classes read it. That's for a different PR though.

Reply:
> We should probably add support for `Body::noun` and let the various classes read it. That's for a different PR though.

Okay, happy to make this change in a different PR.  I'm not sure I understand what you mean but I think if we chat a bit in discord I can learn what you mean and open an appropriate PR.